projects/storefrontlib/shared/components/captcha/mock-captcha/mock-captcha.service.ts

```diff
@@ -78,15 +87,21 @@ export class MockCaptchaService extends CaptchaService {
   renderCaptcha(renderParams: RenderParams): Observable<string> {
     if (renderParams.element instanceof HTMLElement) {
       // Reset checkbox state before rendering
-      this.checkbox.disabled = false;
-      this.checkbox.checked = false;
-      this.label.textContent = "I'm not a robot";
+      this.renderer.setProperty(this.checkbox, 'disabled', false);
+      this.renderer.setProperty(this.checkbox, 'checked', false);
+      this.renderer.setProperty(this.label, 'textContent', "I'm not a robot");
       this.token = '';
       this.retVal = new Subject<string>();
 
-      renderParams.element.appendChild(this.container);
+      this.renderer.appendChild(renderParams.element, this.container);
     }
 
     return this.retVal.asObservable();
   }
+
+  ngOnDestroy() {
+    if (this.removeCheckboxListener) {
+      this.removeCheckboxListener();
```

Comment:
any method needed?, as we bind at initialize() time, I guess we need to unbind the event ?